Source/UI/NewUI/PaymentForm/View/PaymentHeaderView.swift

```diff
@@ -71,14 +72,12 @@ public final class PaymentHeaderView: UIView {
   }
 
   private func updateIconsStack() {
-    guard let icons = style?.schemeIcons else {
-      iconsStackView.isHidden = true
-      return
-    }
-    iconsStackView.isHidden = false
-    for image in icons where image != nil {
+    iconsStackView.isHidden = style?.showAcceptedCardsList ?? true
```

Comment:
```suggestion
    iconsStackView.isHidden = style?.showAcceptedCardsList != true
```

Reply:
I don't think this right solution

🟢 expected result = true  

- if style?.showAcceptedCardsList = nil 
`nil != true`  
then result  = true 

🔴 expected result = true

- if style?.showAcceptedCardsList = true 
`true != true`  
then result  = false 

🔴 expected result = false

- if style?.showAcceptedCardsList = false 
`true != false`  
then result  = true

Reply:
I think that is exactly the right solution :)

if style?.showAcceptedCardsList = nil
nil != true
then result = true `=> Hide icon stack view when showAcceptedCardsList is nil`

if style?.showAcceptedCardsList = true
true != true
then result = false `=> Show icon stack view when showAcceptedCardsList is true`

if style?.showAcceptedCardsList = false
false != true
then result = true `=> Hide icon stack view when showAcceptedCardsList is false`

Reply:
Look at it differently, 
> if style says **SHOW**, UI responds **HIDE**
Basically the assignment to `.isHidden` should be the opposite of `show`:
`view.isHidden = !showSomething`.

And as you pointed out, my suggestion also covers correctly the nil case.

Reply:
I got confused more :D  I think`?? true` is  more simple.

Reply:
That is one of the issues of our acceptance of excessive optionality. I think we can definitely improve on that across the new code.
***
`?? true` may seem clearer for now, but your code is currently using the opposite logic. 

If it wants to show card list its hiding the list. 
If it doesn't want to show card list it will show the list.
I am not sure whether you can negate an optional assignment with that syntax without adding some form of comparison. And when you do, my suggestion might make more sense :)

Reply:
@ehab-al-cko are you planning to make any changes on this thread ? Otherwise all looks good.

Reply:
I would like to keep it as default value for now :)

Reply:
I am not sure this is about preference, its about behaviour at this stage. Your current code does the wrong thing, I'm not sure if you've ran it?

Reply:
Naming `isHidden = show` cause confusion. I have choose better naming.

Reply:
I don't agree with the new change, I have previously suggested [being positive](https://github.com/checkout/frames-ios/pull/236#discussion_r938716528).

We want the language to be clear to consumer, and at our implementation level we absolutely should think clearly what we're doing to map correctly. This is what my comment was for, fixing implementation to map correctly.

Reply:
Thanks for suggestion. I don't agree this is negative approach. :) i would like to keep it for avoid confusion.